### ISSUE TITLE
default_locale did not work

### ISSUE THREAD

**Issue by jaysneg** [amberframework/citrine-i18n#2]
settings.default_locale = "ua" did not work

**Comment by faustinoaq** [comment]
Maybe related to https://github.com/amberframework/amber/issues/776 @docelic WDYT? :sweat_smile:

**Comment by docelic** [comment]
Hello Jaysneg!

@faustinoaq based on the description, I assume that the language files were not found, so the setting of "ua" was ignored and defaulted back to "en".

But sure, let's handle #776 first and then we can see. I will submit a PR for #776 maybe even today.

Thanks!

**Comment by faustinoaq** [comment]
@docelic Oh, ok :+1: BTW, WDYT about moving `locales/` directory from `src/` to `config/` like rails does?

**Comment by docelic** [comment]
Sure, sounds good. Will be another PR. I'm working on it, sorry for the previous delay. Thanks!

**Comment by docelic** [comment]
@jaysneg as a suggestion for you, please edit lib/citrine-i18n/src/citrine-i18n/handler.cr in your project and uncomment the lines that say:

   13           #Amber.logger.debug "Languages available: #{languages.to_s}"
   14           #Amber.logger.debug "Language chosen: #{::I18n.locale}"

Then, please observe the logs while you visit the page, and see whether "ua" appears in the list of available languages. Our further advice will depend on whether it does or not.

Thanks!

**Comment by jaysneg** [comment]
it works if i put this code to slang template
if i didn't do this some of locales didn't work
- I18n.locale = "ua"

**Comment by jaysneg** [comment]
@docelic if i add this to initializers/i18n.cr
I18n.available_locales

it returns ["en", "ru", "ua"]

**Comment by jaysneg** [comment]
it looks like it using javascript for setting this values

**Comment by jaysneg** [comment]
01:34:57 Server     | Languages available: en-US,en;q=0.9,ru;q=0.8,uk;q=0.7
01:34:57 Server     | Language chosen: en

**Comment by jaysneg** [comment]
but in src/locales 3 files
en.uml
ru.yml
ua.yml

**Comment by docelic** [comment]
@jaysneg right, the "Languages available" is somewhat misleading title. The values displayed there (along with q= preferences) come from the HTTP header Accept-Language which is sent by the browser.

**Comment by docelic** [comment]
@jaysneg If I understand it right, the issue here is that the browser is instructing the server to prefer English over other languages, and since "en" is on the list of locales found by Amber, it is naturally chosen instead of the default_locale or other choices.

(This may be somewhat inconvenient for users because users are typically given a choice of language in the web app, and are not used to the model where language preferences are taken from their browser's header Accept-Language.)

Could someone confirm that that's what is happening and advise which path should be taken to solve it?

**Comment by jaysneg** [comment]
my iphone default system language Ukrainian 

03:31:25 Headers    | Accept-Language: ["uk-ua"]

file name uk-ua.yml

and it did not change locale to uk-ua 
site still on EN locale

**Comment by jaysneg** [comment]
03:44:10 Headers    | User-Agent: ["Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_4) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/11.1 Safari/605.1.15"]
03:44:10 Headers    | Accept-Language: ["uk-ua"]

in Safari

**Comment by jaysneg** [comment]
03:44:59 Headers    | Accept-Encoding: ["gzip, deflate, br"]
03:44:59 Headers    | Accept-Language: ["en-US,en;q=0.9,ru;q=0.8,uk;q=0.7"]
in Chrome

system on Ukrainian

**Comment by docelic** [comment]
@jaysneg ok, thanks for all the input. We're working on merging https://github.com/amberframework/amber/pull/814 and then we can continue on this issue.

**Comment by docelic** [comment]
Hey @jaysneg, the above issue was just merged so as soon as I get a chance I'll re-test the functionality from this issue and report back.

Thanks!
